app/src/main/java/com/firebase/uidemo/database/realtime/FirebaseDbPagingActivity.java

```diff
@@ -0,0 +1,153 @@
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.firebase.ui.database.paging.FirebasePagingOptions;
+import com.firebase.ui.database.paging.FirebaseRecyclerPagingAdapter;
+import com.firebase.ui.database.paging.listener.StateChangedListener;
+import com.firebase.uidemo.R;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.Query;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class FirebaseDbPagingActivity extends AppCompatActivity {
+
+    private final String TAG = "PagingActivity";
+
+    @BindView(R.id.paging_recycler)
+    RecyclerView mRecycler;
+
+    @BindView(R.id.paging_loading)
+    ProgressBar mProgressBar;
+
+    private Query mQuery;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_database_paging);
+        ButterKnife.bind(this);
+
+        mQuery = FirebaseDatabase.getInstance().getReference().child("posts");
+
+        setUpAdapter();
+    }
+
+    private void setUpAdapter() {
+
+        //Initialize Paging Configurations
+        PagedList.Config config = new PagedList.Config.Builder()
+                .setEnablePlaceholders(false)
+                .setPrefetchDistance(5)
+                .setPageSize(30)
+                .build();
+
+        //Initialize Firebase Paging Options
+        FirebasePagingOptions<Post> options = new FirebasePagingOptions.Builder<Post>()
+                .setLifecycleOwner(this)
+                .setQuery(mQuery, config, Post.class)
+                .build();
+
+        //Initializing Adapter
+        FirebaseRecyclerPagingAdapter<Post, PostViewHolder> mAdapter =
+                new FirebaseRecyclerPagingAdapter<Post, PostViewHolder>(options) {
+                    @NonNull
+                    @Override
+                    public PostViewHolder onCreateViewHolder(@NonNull ViewGroup parent,
+                                                             int viewType) {
+                        View view = LayoutInflater.from(parent.getContext())
+                                .inflate(R.layout.item_post, parent, false);
+                        return new PostViewHolder(view);
+                    }
+
+                    @Override
+                    protected void onBindViewHolder(@NonNull PostViewHolder holder,
+                                                    int position,
+                                                    @NonNull String key,
+                                                    @NonNull Post model) {
+                        holder.bind(model);
+                    }
+                };
+
+        mRecycler.setLayoutManager(new LinearLayoutManager(this));
+        mRecycler.setAdapter(mAdapter);
+
+        mAdapter.setStateChangedListener(new StateChangedListener() {
+            @Override
```

Comment:
But in that case we will not able to show `databaseError` in `onError`

Reply:
Can I override two methods in adapter class like..
```
onLoadingStateChanged(LoadingState loadingState)
onError(DatabaseError databaseError)
```
So that it will display information about database errors ?

Reply:
Ah yes that seems the same as this feature request:
https://github.com/firebase/FirebaseUI-Android/issues/1592

I am fine with adding a second callback!  That will solve this issue at the same time.

Reply:
Okay. I Have added it in a recent commit.